src/crypto.js

```diff
@@ -88,7 +98,7 @@ export default function makeCryptoWith(opts) {
       const [ serializedInput, iv, key ] = results;
       const cipher = crypto.createCipheriv(CIPHER_ALGORITHM, key, iv);
 
-      if(aad !== undefined) {
+      if(_validateAADType(aad)) {
```

Comment:
I find using `_valaidateAADType` to throw an error and return whether it's specified to be rather confusing. Can we change it to only do the validation and then check to see if we should specify it in a different step? Something like the following:

```
function _validateAAD(aad) {
  if (aad == null) {
     return;
  }

  if (typeof aad !== 'string') {
    throw new Error('AAD must be a string');
  }

  if (aad.length === 0) {
    throw new Error('AAD cannot be an empty string');
  }
}

_validateAAD(aad); // at the very top of encrypt and decrypt

if(aad != null) {
  cipher.setAAD(Buffer.from(aad, 'utf8'));
}
```

Reply:
@kobelb I don't have strong opinions either way. This way is fine by me.